### ISSUE TITLE
npx init with "--package-name" makes Android application crash on runtime

### ISSUE THREAD

**Issue by jvaronenka** [facebook/react-native#42280]
### Description

If you initialise an application with "npx init" and provide a package name with "--package-name", then the built Android app will crash on startup.

No such issue exists on iOS.
No such issue on 0.72 versions.

I have narrowed this down to how `MainActivity.kt` is generated. It generates the following function:
```
override fun getMainComponentName(): String = "com.mycompany.MyApp"
```

However, it should be:
```
override fun getMainComponentName(): String = "MyApp"
```

Manually changing that function and rebuilding resolves the issue.

### Steps to reproduce

1. `npx react-native@0.73.2 init MyApp --version 0.73.2 --package-name com.mycompany.MyApp`
2. `cd MyApp/android`
3. `./gradlew assembleRelease`
4. Launch app/build/outputs/apk/release/app-release.apk

### React Native Version

0.73.2

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
n/a
```

### Stacktrace or Logs

```text
n/a
```

### Reproducer

n/a

**Comment by jvaronenka** [comment]
I can't add a reproducer, but I have added reproduction steps. Is that sufficient?

**Comment by jvaronenka** [comment]
Duplicate of https://github.com/react-native-community/cli/issues/2205

**Comment by cortinico** [comment]
Yup this will be fixed in the upcoming patch of 0.73
